### ISSUE TITLE
Default generated js libraries cannot be dynamically installed & required.

### ISSUE THREAD

**Issue by sebastiandg7** [nrwl/nx#14933]
### Current Behavior

The current generated `@nrwl/nx-plugin` plugins contain the package's index file in the `src/index.ts` directory. A bug with NodeJS nodejs/node#44663 prevents from a running process to install and invoke a package if it's index file is not in the package's root. Read more in qwikifiers/qwik-nx#44.

### Expected Behavior

Generated projects should contain the `index.ts` located at the package's root. That way plugins can be invoked in scenarios like this one:

```typescript
await ensurePackage(tree, '@nxkit/playwright', nxKitVersion);
const { projectGenerator } = await import('@nxkit/playwright'); // <= throws
```

### Steps to Reproduce

Run the `qwik-nx` plugin preset generator from version 0.9.0

`pnpx create-nx-workspace --preset qwik-nx`

### Nx Report

```shell
N/A as error happens in the workspace creation.
```

### Failure Logs

```shell
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn pnpm ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn pnpm',
  path: 'pnpm',
  spawnargs: [
    'exec',
    'nx',
    '--nxWorkspaceRoot="C:/dev/workspace/open-source"',
    'g',
    'qwik-nx:preset',
    '--no-a',
    '--no-g',
    '--$0=C:\\Users\\MyUser\\AppData\\Local\\pnpm\\store\\v3\\tmp\\dlx-19560\\node_modules\\create-nx-workspace\\bin\\index.js',
    '--packageManager=pnpm',
    '--defaultBase=main'
  ]
}

Node.js v18.7.0
 ERROR  Command failed with exit code 1: C:\Users\MyUser\AppData\Local\pnpm\store\v3\tmp\dlx-19560\node_modules\.bin\create-nx-workspace.CMD --preset qwik-nx

pnpm: Command failed with exit code 1: C:\Users\MyUser\AppData\Local\pnpm\store\v3\tmp\dlx-19560\node_modules\.bin\create-nx-workspace.CMD --preset qwik-nx
    at makeError (C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:22852:17)
    at handlePromise (C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:23423:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.handler [as dlx] (C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:202440:7)
    at async C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:209882:21
    at async main (C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:209853:34)
    at async runPnpm (C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:210079:5)
    at async C:\Users\MyUser\AppData\Roaming\nvm\v18.7.0\node_modules\pnpm\dist\pnpm.cjs:210071:7
```

**Comment by sebastiandg7** [comment]
Official Nx plugins have their `index.ts` file in the package root, so it would be consistent to generate custom plugins with that same structure.

**Comment by dmitry-stepanenko** [comment]
Since this is a bug with node, maybe as an alternative it is worth just adding a check within `ensurePackage` function, that would print an Nx warning message, if installed package does not have `index.js` in its root

**Comment by sebastiandg7** [comment]
@dmitry-stepanenko That could be added as an extra prevention for plugins using **"**wrongfully built packages**"** (you know, a node problem). Yet, it could be annoying for plugins just installing them and not immediately invoking an executor/generator which is most `init` generators case.

Do you think is worth adding a migration so existing plugins get their `./src/index.ts` moved to `./index.ts` plus paths in `index.ts` content, `package.json`, `project.json` and `tsconfig.lib.json` changes ?

**Comment by AgentEnder** [comment]
Hey @sebastiandg7 and @dmitry-stepanenko ! Thanks for the excellent investigative work and report on this one.. I looked over the associated issue on the node repo and its definitely something that needs to be addressed, but it should probably be more holistically fixed by updating the logic in `@nrwl/js:library`.

This is since we encourage folks to build publishable libraries with it, and the nx-plugin generator piggybacks on it. The nx-plugin generator will also need to be updated, but we should address the root cause first.

On our side of things I've talked with @nartc and we should be able to address it at some point this week. Its definitely an interesting issue though, and has some layers involved with `tsc` and source root settings also being involved.

Thanks for the collaboration on this, and please stay involved! After the @nrwl/js:library generator is updated, changes similar to the ones in the closed PR (#14941) will be needed.
